### ISSUE TITLE
OSX, dynamic compile: link errors with 1.8.0 package

### ISSUE THREAD

**Issue by JohanEngelen** [ldc-developers/ldc#2618]
With a simple test case (e.g. the `dynamiccompile/array.d` test), `ldc2 -enable-dynamic-compile -run %s` does not work on OSX with the 1.8.0 release package on github.
The error:
```
❯ bin/ldc2 -enable-dynamic-compile -run dyntest.d
ld: warning: ignoring file /Users/johan/ldc/ldc2-1.8.0-osx-x86_64/lib/libldc-jit-rt.a, file was built for archive which is not the architecture being linked (x86_64): /Users/johan/ldc/ldc2-1.8.0-osx-x86_64/lib/libldc-jit-rt.a
Undefined symbols for architecture x86_64:
  "__D3ldc15dynamic_compile16CompilerSettings6__initZ", referenced from:
      __Dmain in dyntest-8a28465.o
...
```

Copying self-built `libldc-jit-rt.a` onto the one of the 1.8.0 release package, linking works. But execution fails with the same issue found on our buildserver (https://github.com/ldc-developers/ldc/pull/2532) and reported in https://github.com/ldc-developers/ldc/issues/2615 .
```
❯ bin/ldc2 -enable-dynamic-compile -run dyntest.d
dyld: Library not loaded: @rpath/libldc-jit.78.dylib
  Referenced from: /var/folders/5b/m2dlbc8s071fq8m0tbsyrj6h0000gn/T/dyntest-1aeaea6-e9eb61
  Reason: image not found
```

So two issues here:
- [ ] release package somehow builds an incompatible `libldc-jit-rt.a` (incompatible with my XCode? `Xcode 9.2  Build version 9C40b`)
- [ ] rpath should be set for the jit library linked with `-enable-dynamic-compile`. Similar to what is done for e.g. ASan.

**Comment by kinke** [comment]
The 2nd issue is fixed in #2659 (setting rpath when linking with the shared ldc-jit lib).

I'm surprised by the first issue though; glancing over the CMake script, the archive should be 64-bits exclusive for `MULTILIB=ON` too.

**Comment by kinke** [comment]
```
# intermediate libldc-jit-rt.a:
$ lipo -info libldc-jit-rt.a 
input file libldc-jit-rt.a is not a fat file
Non-fat file: libldc-jit-rt.a is architecture: x86_64

# installed copy:
fatal error: /Applications/Xcode-9.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: archive with no architecture specification: libldc-jit-rt.a (can't determine architecture for it)
```

**Comment by kinke** [comment]
It appears to be that `ranlib` issue on OSX again; using `RT_ARCHIVE_WITH_LDC=OFF` yields a valid CMake-installed `libldc-jit-rt.a` (but then the LTO integration tests fail... sigh).

**Comment by kinke** [comment]
Fixed by #2659.
